app/models/asset.rb

```diff
@@ -245,69 +217,51 @@ def printable_object(username = 'unknown')
       :top_line => TokenUtil.human_barcode(barcode),
       :middle_line => kit_type,
       :bottom_line => info_line
-      }
-    }
+    } }
   end
 
   def kit_type
     activities_affected&.last&.kit&.kit_type&.abbreviation
   end
 
   def position_value
-    val = facts.map(&:position).compact.first
+    val = facts.filter_map(&:position).first
     return "" if val.nil?
-    "_#{(val.to_i+1).to_s}"
+
+    "_#{(val.to_i + 1).to_s}"
   end
 
   def info_line
-    "#{class_name} #{aliquot} #{position_value}".strip
+    "#{purpose_name} #{aliquot} #{position_value}".strip
   end
 
-  def class_name
-    purposes_facts = facts.with_predicate('purpose')
-    if purposes_facts.count > 0
-      return purposes_facts.first.object
-    end
-    return ''
+  def purpose_name
+    # @todo: Falling back to an empty string, rather than nil feels a bit risky, but this maintains earlier behaviour.
+    # (We don't have any cases of facts with predicate purpose and value NULL)
+    facts.with_predicate('purpose').first&.object || ''
```

Comment:
This was previously hitting the database twice, once to check how many we had, then again to fetch the first record. Now we just hit it the once.